src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -102,23 +115,21 @@ def tryConnect(self, host, port):
                 s.settimeout(None)
                 s.close()
 
-            i += 1
+            if connected:
```

Comment:
Why have a `connected` var at all? you could just return True after `s.connect` and have a return False at the end.

Reply:
I'm not an expert in the way how Python actually deals with the sockets, possibly it'll close the with garbage collection, but to prevent resource leaking I'd prefer to explicitly close the socket. Also - John did it! :)